AutoCollection/WebSnippet.ts

```diff
@@ -0,0 +1,146 @@
+        }
+        fs.readFile(snippetPath, function (err, snippet) {
+            if (err) {
+                Logging.warn("Failed to load AI Web snippet. Ex:" + err);
+            }
+            WebSnippet._snippet = snippet.toString().replace("INSTRUMENTATION_KEY", client.config.instrumentationKey);
+        });
+    }
+
+    public enable(isEnabled: boolean) {
+        this._isEnabled = isEnabled;
+
+        if (this._isEnabled && !this._isInitialized) {
+            this._initialize();
+        }
+    }
+
+    public isInitialized() {
+        return this._isInitialized;
+    }
+
+    private _initialize() {
+        this._isInitialized = true;
+        const originalHttpServer = http.createServer;
+        http.createServer = (requestListener?: (request: http.IncomingMessage, response: http.ServerResponse) => void) => {
+            const originalRequestListener = requestListener;
+            if (originalRequestListener) {
+                requestListener = (request: http.IncomingMessage, response: http.ServerResponse) => {
+                    // Patch response write method
+                    let originalResponseWrite = response.write;
+                    let isEnabled = this._isEnabled;
+                    response.write = function wrap(a: Buffer | string, b?: Function | string, c?: Function | string) {
+                        if (isEnabled) {
+                            if (WebSnippet.ValidateInjection(response, a)) {
+                                arguments[0] = WebSnippet.InjectWebSnippet(response, a);
+                            }
+                        }
+                        return originalResponseWrite.apply(response, arguments);
+                    }
+                    // Patch response end method for cases when HTML is added there
+                    let originalResponseEnd = response.end;
+
+                    response.end = function wrap(a?: Buffer | string | any, b?: Function | string, c?: Function) {
+                        if (isEnabled) {
+                            if (WebSnippet.ValidateInjection(response, a)) {
+                                arguments[0] = WebSnippet.InjectWebSnippet(response, a);
+                            }
+                        }
+                        originalResponseEnd.apply(response, arguments);
+                    }
+
+                    originalRequestListener(request, response);
+                }
+            }
+            return originalHttpServer(requestListener);
+        }
+    }
+
+    /**
+     * Validate response and try to inject Web snippet
+     */
+    public static ValidateInjection(response: http.ServerResponse, input: string | Buffer): boolean {
+        if (response && input) {
+            // insert snippet if only response returns 200
+            if (response.statusCode != 200) return false;
+            let contentType = response.getHeader('Content-Type');
```

Comment:
mixed use of single-quotes and double-quotes.  Can you make the usage consistent across the project?  Check with Hector on what the standard is.

Reply:
updated in helper methods